Storage/rest/advanced_hypermedia.h

```diff
@@ -34,15 +34,21 @@ SOFTWARE.
 #include "hypermedia.h"
 #include "sfinae.h"
 
-CURRENT_STRUCT(AdvancedHypermediaRESTRecordResponse) {
+namespace current {
+namespace storage {
+namespace rest {
+
+CURRENT_STRUCT_T(AdvancedHypermediaRESTRecordResponse) {
+  CURRENT_FIELD(success, Optional<bool>);
```

Comment:
In Hypermedia (REST), there is a resource, and a collection of resources. So, I'd expect:
* for a request querying a resource, a ResourceResponse:
```
// success
{ success: true, data: { url: "/resource1", data: { foo: "bar1" } } }

// error
{ success: false, error: { name: "ResourceNotFoundError", details: { id: "1" } } }
```
* for a request querying a collection of resources, a CollectionOfResourcesResponse: 
```
// success
{ success: true, data: [
  { url: "/resource1", data: { foo: "bar1" } },
  { url: "/resource2", data: { foo: "bar2" } }
] }

// error
{ success: false, error: { name: "CollectionNotFoundError" } }
```